### PR TITLE
Mini-Challenge 1 & 2

### PR DESCRIPTION
**Pull request by jonasaky**
## This PR covers the following:

Mini-Challenge 1 : Core Concepts and Styling

1. The header is rendered correctly.
2. A list of videos obtained from a mock file is displayed in the Home View.
3. CSS styles are applied correctly using the styled-components approach.

Mini-Challenge 2 : Intro to Testing 

1. Find any sub-routine, extract that logic into a separate file and test it out.
2. Run test coverage and save results into a .txt file at ./ - root level. Can you make the total coverage to be above of 60%?

### CODE REVIEW

File: src/components/NavBar/NavBar.component.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            
+            <Styled.CheckBoxWrapper >
+                <Styled.CheckBox id="checkbox" type="checkbox" />
+                <Styled.CheckBoxLabel htmlFor="checkbox" />
+            </Styled.CheckBoxWrapper>
+
+            <Styled.LoginButton >Login</Styled.LoginButton>
+        </Styled.NavBar>
+    )
+}
```
Comment (PacoMojica) [inline src/components/NavBar/NavBar.component.jsx]: This looks great, you can go one step further and do something like:

```jsx
import {
  NavBar,
  SearchField,
  CheckBoxWrapper,
 ...
} from './Styled';
```

this makes the code more readable

File: src/components/NavBar/styled.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      width: 18px;
+      height: 18px;
+      margin-left: 21px;
+      transition: 0.2s;
+    }
+  }
+`;
+
+const Styled = { NavBar, SearchField, LoginButton, CheckBoxWrapper, CheckBox, CheckBoxLabel };
+export default Styled;
```
Comment (PacoMojica) [inline src/components/NavBar/styled.js]: the last 2 lines are uncommon ways of achieving what you want to do, the common way of writing this is something like:

```jsx
...
export const CheckBoxLabel = styled.label`
...
`;

export const CheckBox = styled.input`
...
`;
...
```

with this you can delete lines 73 & 74, and when you want to use any component defined here you can do:

```jsx
import { ... CheckBoxLabel, CheckBox, ... } from './Styled';
```

File: src/components/NavBar/tests/NavBar.spec.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    });
+
+    it('contains toggle button to switch theme', () => {
+        render(<NavBar />);
+        expect(screen.getByRole('checkbox')).toBeTruthy();
+    });
+
+    it('contains a login button', () => {
+        render(<NavBar />);
+        expect(screen.getByText('Login')).toBeTruthy();
```
Comment (PacoMojica) [inline src/components/NavBar/tests/NavBar.spec.js:18]: Good test! an alternative to `toBeTruthy` is `toBeInTheDocument`

File: src/components/VideoCard/styled.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+`;
+
+const VideoTitle = styled.p`
+    background-color: rgba(0, 122, 122, 0.23);
+    z-index: 1;
+    height: 180px;
+`;
+
+const Styled = { VideoCard, VideoImg, VideoTitle };
+export default Styled;
```
Comment (PacoMojica) [inline src/components/VideoCard/styled.js]: Same comment as before, you can use the 
```jsx
export const MyComponent = ....
```

this way you can omit the creation of the object and the default export, when you use `export const MyComponent...` JS creates the object for you

**Review by PacoMojica — COMMENTED** [review]
Overall you did great progress 👍, all I found were some minor details, keep the good work coming 😀

File: src/components/NavBar/NavBar.component.jsx (hunk 1 above)
Comment (jonasaky) [reply src/components/NavBar/NavBar.component.jsx]: great! thanks for the suggestion

**Comment by jonasaky** [comment]
Thanks for your review, @PacoMojica ! 
I did the changes as you suggested and now the code looks great.

**Pull request merged**